src/passes/Metrics.cpp

```diff
@@ -62,23 +61,26 @@ struct Metrics
     counts["[tags]"] = imports.getNumDefinedTags();
     counts["[exports]"] = module->exports.size();
     counts["[tables]"] = imports.getNumDefinedTables();
-    // add memory and table
-    if (module->memory.exists) {
-      Index size = 0;
-      for (auto& segment : module->memory.segments) {
-        size += segment.data.size();
-      }
+
+    // add memory
+    walkMemory(&module->memory);
+    Index size = 0;
+    for (auto& segment : module->dataSegments) {
+      walkDataSegment(segment.get());
+      size += segment->data.size();
+    }
+    if (!module->dataSegments.empty()) {
```

Comment:
```suggestion
    if (module->memory.exists) {
```
I think this might remove the diff on the metrics test?

Reply:
(And after adding multiple memory support, the line could be
```cpp
    if (!module->memories.empty()) {
```
which would be symmetric to the table code below.)

Reply:
Sweet, thanks for the idea. I changed the if-check per your suggestion but the file `test/passes/O3_low-memory-unused_metrics.txt` didn't revert back.

Reply:
Ok, I guess I'm missing something here... let's leave this as-is then (with multi-memory it'll change anyhow).